packages/awslint/lib/rules/api.ts

```diff
@@ -3,143 +3,220 @@ import { ConstructReflection } from './construct';
 import { CoreTypes } from './core-types';
 import { Linter } from '../linter';
 
-const EXCLUDE_ANNOTATION_REF_VIA_INTERFACE = '[disable-awslint:ref-via-interface]';
+const EXCLUDE_ANNOTATION_REF_VIA_INTERFACE =
+  '[disable-awslint:ref-via-interface]';
+
+const EXCLUDE_ANNOTATION_REF_VIA_REF_INTERFACE =
+  '[disable-awslint:ref-via-ref-interface]';
```

Comment:
I considered wrapping them both into a single rule, but found it too confusing and complicated.
Now we have two rules:
- `ref-via-interface` guarantees that we are not expecting concrete instances of constructs
- `ref-via-ref-interface` guarantees that we prefer `ICfnWhatever` reference interfaces over `IWhatever` interfaces